### ISSUE TITLE
longer arrays

### ISSUE THREAD

**Issue by zkessin** [jsverify/jsverify#205]
When you generate an array in a standard property it seems to top out at a size of about 6, this may miss many cases that are easier to spot in a longer array. For example, if a bug requires an element to be in an array 3 times, it is much more likely to happen if the array has a length of 20 or 60 elements or more

**Comment by jvanbruegge** [comment]
You can specify the test size with the options. The default test size of 50 is very small, you can try setting it to something like 200, this should generate bigger arrays too

**Comment by glebec** [comment]
For some reason the array generator is actually based on the _log_ of the size.

https://github.com/jsverify/jsverify/blob/c03a67efe525e79ec37b3f314fc9f6a30182be70/lib/generator.js#L218

This makes it very hard to have nontrivial array lengths. I don't understand the reason behind this.

**Comment by dubzzz** [comment]
@glebec I believe that the main reason for this is that the shrinking of arrays in jsverify is very slow.. too slow. I personally open sourced my own property based js lib for that reason

**Comment by glebec** [comment]
The obvious follow-up in that case would be "can we make shrinking of arrays faster"? I haven't looked deeply at the implementation of array shrinks to be able to answer this offhand, but it could be an important part of addressing this issue if it is possible.

**Comment by dubzzz** [comment]
Yep maybe by copying the shrinker of arrays defined in [fast-check](https://github.com/dubzzz/fast-check) :P

**Comment by dubzzz** [comment]
That being said I did try a faster implementation of array shrinker for jsverify before starting fast-check, see https://github.com/dubzzz/jsverify-commands/blob/master/src/arbitrary/commands.js

**Comment by glebec** [comment]
Thanks for the interesting links, will read further later. I happen to be writing a Dev.to article on JSVerify at the moment and my introductory example was sorting of arrays of integers – just wanted to let you know that this is a coincidence, not attempting to rip off your Fast-Check documentation! 😅

**Comment by dubzzz** [comment]
Profile updated - check my profile for personal message
